### ISSUE TITLE
which are the steps to add pt-br (brazilian portuguese) to the rasa?

### ISSUE THREAD

**Issue by fafg** [golastmile/rasa_nlu#195]
I pretend to work with pt-br =)
I have a tokenizer (NLPNET) for pt-br o/

**Comment by PHLF** [comment]
Here is the process I followed for French (thanks to @amn41 and @davisking):

1. Get a ~clean language corpus (a wikipedia dump works) as a set of text files
2. Build and run [MITIE wordrep tool](https://github.com/mit-nlp/MITIE) on your corpus. This can takes several hours/days depending on your dataset and your workstation. You'll need something like 128GB of RAM for wordrep to run -Yes that's alot: try to extend your swap-
3. Set the path of your new *total_word_feature_extractor.dat* as value of the **mitie_file** parameter in config.mitie.json

Your good to go using MITIE backend. In regards to spaCy, you'll need a tokenizer (MITIE is all inclusive) and the procedure seems more complex.

**Comment by tmbo** [comment]
@PHLF If you don't mind, I'd create a new page on the documentation adding these thoughts as a reference.

**Comment by tmbo** [comment]
The help page can be found here: 

https://rasa-nlu.readthedocs.io/en/latest/languages.html

**Comment by pdesgarets** [comment]
Hey @PHLF , is the french total_word_feature_extractor.dat available somewhere ? That would save me a lot of time...
Thanks !

**Comment by PHLF** [comment]
I can't help you with that: as I built it with my current company's resources, the file doesn't belong to me. Moreover I used french wikipedia as a corpus, so as I said I doubt the language model is very robust to bad user input/typos. Nevertheless if you want to do the same process to try for yourself, I can be more specific:
- [A fresh dump of french wiki](https://dumps.wikimedia.org/frwiki/latest/frwiki-latest-pages-articles.xml.bz2)
- [The tool to transform it in a usable input for MITIE](https://github.com/attardi/wikiextractor)
  - [How to remove the doc tag](https://github.com/attardi/wikiextractor/issues/120)
- Put all the files you obtained in the same folder: this will be MITIE input folder (see below)
- If you don't have enough RAM (~128GB), increase your swap. I had as little as 32GB RAM on my workstation, so even if swap is extremely slow, it should work.
- [Compile and use MITIE wordrep with your fresh wiki dump](https://github.com/mit-nlp/MITIE/tree/master/tools/wordrep) `wordrep -e /MITIE/input/folder`. Be patient as it may take a few days without outputting anything (It took 2.5 days for me using an high end Intel i7 quadcore CPU).

Following this recipe, you'll get a total_word_features_extractor.dat

Hope this help
